sidecar/discovery/discovery.go

```diff
@@ -159,3 +198,247 @@ func filterInstances(instances []*api.ServiceInstance, tags []string) []*api.Ser
 
 	return filtered
 }
+
+// getRegistration
+func (d *Discovery) getClusters(w rest.ResponseWriter, req *rest.Request) {
+	//clusterName := req.PathParam(routeParamServiceClusterName)
+	//
+	//nodeName := req.PathParam(routeParamServiceNodeName)
+
+	instances, err := d.discovery.ListInstances()
+	if err != nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+
+	ruleSet, err := d.rules.ListRules(&api.RuleFilter{})
+	if err != nil {
+		w.WriteHeader(http.StatusServiceUnavailable)
+		return
+	}
+
+	resp := Clusters{
+		Clusters: buildClusters(instances, ruleSet.Rules, d.tlsConfig),
+	}
+
+	w.WriteHeader(http.StatusOK)
+	w.WriteJson(&resp)
+}
+
+func buildClusters(instances []*api.ServiceInstance, rules []api.Rule, tlsConfig *envoy.SSLContext) []envoy.Cluster {
+
+	clusterNames := make(map[string]struct{})
+	for _, instance := range instances {
+		clusterName := envoy.BuildServiceKey(instance.ServiceName, instance.Tags)
+		clusterNames[clusterName] = struct{}{}
+		// Need a default cluster for every service
+		clusterNames[instance.ServiceName] = struct{}{}
+	}
+
+	// TODO ?
+	envoy.SanitizeRules(rules)
+	rules = AddDefaultRouteRules(rules, instances)
+
+	backends := make(map[string]*api.Backend)
+	for _, rule := range rules {
+		if rule.Route != nil {
+			for _, backend := range rule.Route.Backends {
+				key := envoy.BuildServiceKey(backend.Name, backend.Tags)
+				// TODO if two backends map to the same key, it will overwrite
+				//  and will lose resilience field options in this case
+				backends[key] = &backend
+			}
+		}
+	}
+
+	clusters := make([]envoy.Cluster, 0, len(clusterNames))
+	for name := range clusterNames {
+		cluster := envoy.Cluster{
+			Name:             name,
+			ServiceName:      name,
+			Type:             "sds",
+			LbType:           "round_robin", //TODO this needs to be configurable
```

Comment:
`master` supports configurable load balancing right now, right? Can you update the logic here appropriately?